orchestrator/batch_request.go

```diff
@@ -80,64 +80,50 @@ func (r *batchRequester) getUnbatchedFeesByToken(ctx context.Context, injective
 	return unbatchedFees, nil
 }
 
-func (r *batchRequester) requestBatchCreation(
-	ctx context.Context,
-	injective InjectiveNetwork,
-	feed PriceFeed,
-	batchFee *types.BatchFees,
-) {
+func (l *batchRequestLoop) requestBatch(ctx context.Context, fee *types.BatchFees) {
 	var (
-		tokenAddr = eth.HexToAddress(batchFee.Token)
-		denom     = r.tokenDenom(tokenAddr)
+		tokenAddr  = eth.HexToAddress(fee.Token)
+		tokenDenom = l.tokenDenom(tokenAddr)
 	)
 
-	if thresholdMet := r.checkFeeThreshold(feed, tokenAddr, batchFee.TotalFees); !thresholdMet {
-		r.log.WithFields(log.Fields{
-			"denom":          denom,
-			"token_contract": tokenAddr.String(),
-			"total_fees":     batchFee.TotalFees.String(),
-		}).Debugln("skipping underpriced batch")
+	if thresholdMet := l.checkFeeThreshold(tokenAddr, fee.TotalFees); !thresholdMet {
 		return
 	}
 
-	r.log.WithFields(log.Fields{
-		"denom":          denom,
-		"token_contract": tokenAddr.String(),
-	}).Infoln("requesting batch creation on Injective")
+	l.Logger().WithFields(log.Fields{"denom": tokenDenom, "token_contract": tokenAddr.String()}).Infoln("requesting batch on Injective")
 
-	_ = injective.SendRequestBatch(ctx, denom)
+	_ = l.inj.SendRequestBatch(ctx, tokenDenom)
 }
 
-func (r *batchRequester) tokenDenom(tokenAddr eth.Address) string {
-	if cosmosDenom, ok := r.erc20ContractMapping[tokenAddr]; ok {
+func (l *batchRequestLoop) tokenDenom(tokenAddr eth.Address) string {
+	if cosmosDenom, ok := l.erc20ContractMapping[tokenAddr]; ok {
 		return cosmosDenom
 	}
 
 	// peggy denom
+	// todo: in reality, peggy denominators don't have an actual price listing
+	// So it seems that bridge fee must always be inj
 	return types.PeggyDenomString(tokenAddr)
 }
 
-func (r *batchRequester) checkFeeThreshold(
-	feed PriceFeed,
-	tokenAddr eth.Address,
-	totalFees cosmtypes.Int,
-) bool {
-	if r.minBatchFee == 0 {
+func (l *batchRequestLoop) checkFeeThreshold(tokenAddr eth.Address, fees cosmtypes.Int) bool {
+	if l.minBatchFeeUSD == 0 {
 		return true
 	}
 
-	tokenPriceInUSD, err := feed.QueryUSDPrice(tokenAddr)
+	tokenPriceInUSD, err := l.pricefeed.QueryUSDPrice(tokenAddr)
 	if err != nil {
 		return false
 	}
 
+	minFeeInUSDDec := decimal.NewFromFloat(l.minBatchFeeUSD)
 	tokenPriceInUSDDec := decimal.NewFromFloat(tokenPriceInUSD)
-	totalFeeInUSDDec := decimal.NewFromBigInt(totalFees.BigInt(), -18).Mul(tokenPriceInUSDDec)
-	minFeeInUSDDec := decimal.NewFromFloat(r.minBatchFee)
+	totalFeeInUSDDec := decimal.NewFromBigInt(fees.BigInt(), -18).Mul(tokenPriceInUSDDec)
```

Comment:
We can obtain the actual accurate decimals from the alchemy API (for tokens that are supported by their API) and/or fallback to querying the actual ERC-20 decimals function to obtain the accurate number of decimals